orchagent/aclorch.cpp

```diff
@@ -3168,6 +3168,87 @@ void AclOrch::init(vector<TableConnector>& connectors, PortsOrch *portOrch, Mirr
     }
 
 
+
+    // check switch capability of Metadata attribute, action and range.
+    // SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE support and range values.
+    // SAI_ACL_ENTRY_ATTR_ACTION_SET_ACL_META_DATA
+    // SAI_ACL_ENTRY_ATTR_FIELD_ACL_USER_META
+
+    sai_status_t status = SAI_STATUS_SUCCESS;
+    sai_attr_capability_t capability;
+    status = sai_query_attribute_capability(gSwitchId, SAI_OBJECT_TYPE_SWITCH, SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE, &capability);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Could not query ACL_USER_META_DATA_RANGE %d", status);
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_RANGE_CAPABLE] = "false";
+    }
+    else
+    {
+        if (capability.get_implemented)
+        {
+            m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_RANGE_CAPABLE] = "true";
+            sai_attribute_t attrs[1];
+            attrs[0].id = SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE;
+            sai_status_t status = sai_switch_api->get_switch_attribute(gSwitchId, 2, attrs);
+            if (status != SAI_STATUS_SUCCESS)
+            {
```

Comment:
If range is not available, suggest changing `m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_RANGE_CAPABLE]` to false as we don't know what meta can be used.

Reply:
done